python/housinginsights/ingestion/insertion_functions.py

```diff
@@ -0,0 +1,299 @@
+"""
+Insertions_functions.py contains methods necessary for inserting data from 
+one file into another file.
+"""
+
+import os
+import sys
+from uuid import uuid4
+import pandas as pd
+from datetime import datetime
+import numpy as np
+
+# setup some useful absolute paths
+PYTHON_PATH = os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                           os.pardir, os.pardir))
+DATA_RAW_PATH = os.path.join(PYTHON_PATH, os.pardir, 'data', 'raw')
+
+# add to python system path
+sys.path.append(PYTHON_PATH)
+
+from python.housinginsights.sources.mar import MarApiConn
+from python.housinginsights.sources.models.mar import FIELDS as MAR_FIELDS
+from python.housinginsights.sources.models.DCHousing import PROJECT_FIELDS_MAP,\
+    SUBSIDY_FIELDS_MAP
+from python.housinginsights.sources.models.pres_cat import PROJ_FIELDS, \
+    SUBSIDY_FIELDS
+
+
+def project_csv_fix_empty_address_id():
+    """
+    Identifies any rows in the project.csv file with missing 
+    'proj_address_id' and attempts to fix by using latitude/longitude, 
+    proj_x/y, or proj_addre/city/st/zip.
+    
+    :return: 
+    """
+    proj_df = pd.read_csv(os.path.join(DATA_RAW_PATH, 'preservation_catalog',
```

Comment:
@jkwening So unfortunately we'll want to refactor this to not use Pandas by the time this is all done. Pandas forces us to get a bigger server. The only place we use Pandas currently is in the make_draft_json, which is only used in development (not production). 

If this is working we can skip that for now and deal with it later, but wanted to flag it for future development.